addons/im_livechat/models/chatbot_script_step.py

```diff
@@ -0,0 +1,199 @@
+        - step 4 (text): Thank you for visiting our website!
+        -> The welcoming steps will contain steps 1, 2 and 3.
+        Meaning the bot will have a small monologue with himself before expecting an input from the
+        end user.
+
+        This is important because we need to display those welcoming steps in a special fashion on
+        the frontend, since those are not inserted into the mail.channel as actual mail.messages,
+        to avoid bloating the channels with bot messages if the end-user never interacts with it. """
+
+        welcome_steps = self.env['im_livechat.chatbot.script_step']
+        for step in self:
+            welcome_steps += step
+            if step.step_type != 'text':
+                break
+
+        return welcome_steps
+
+    def _is_last_step(self, mail_channel=False):
+        self.ensure_one()
+        mail_channel = mail_channel or self.env['mail.channel']
+
+        # if it's not a question and if there is no next step, then we end the script
+        if self.step_type != 'question_selection' and not self._fetch_next_step(
+           mail_channel.livechat_chatbot_message_ids.mapped('chatbot_question_answer_id')):
+            return True
+
+        return False
+
+    def _process_answer(self, mail_channel, mail_message):
+        """
+        Process user's answer depending on the step_type.
+
+        :param mail_channel:
+        :param mail_message:
+        :return: script step to display next
+        :rtype: 'im_livechat.chatbot.script_step'
+        """
+        self.ensure_one()
+
+        user_raw_answer = html2plaintext(mail_message.body)
+        chatbot_mail_message = self.env['im_livechat.chatbot.mail.message'].search([
+            ('mail_channel_id', '=', mail_channel.id),
+            ('chatbot_step_id', '=', self.id),
+        ], limit=1)
+
+        update_values = {}
+        if self.step_type == 'question_selection':
+            # Update 'chatbot.mail.message' with the user's answer
+            chatbot_question_answer_id = self.answer_ids.filtered(lambda a: a.name == user_raw_answer)
+            if not chatbot_question_answer_id:
+                raise ValidationError(_('"%s" is not a valid answer for this step', user_raw_answer))
+            update_values.update({
+                'chatbot_question_answer_id': chatbot_question_answer_id.id,
+                'chatbot_user_raw_answer': user_raw_answer,
+            })
+
+        elif self.step_type == 'question_email':
+            if not single_email_re.match(user_raw_answer):
+                # if this error is raised, display an error message but do not go to next step
+                raise ValidationError(_('"%s" is not a valid email.', user_raw_answer))
+            update_values['chatbot_user_raw_answer'] = email_normalize(user_raw_answer)
+
+        elif self.step_type == 'question_phone':
+            update_values['chatbot_user_raw_answer'] = user_raw_answer
```

Comment:
Since we most likely don't have a country (_maybe_ from geoip? but does not seem reliable), we can't really validate the phone number.
Indeed, I'm pretty sure the validation absolutely requires a country, and will not work at all without one.
(Could be wrong there, but last time I checked it seemed to be the case).

In addition, it's currently not possible to have a "non-blocking" validation on the frontend side.
It could probably be done, but I'm not sure how, would require some functional guidance and back-and-forth.